### PR TITLE
mitscheme: use texinfo6

### LINKED ISSUE #229784: Build failure: `mitscheme`
### Steps To Reproduce

1. build `mitscheme` from latest master

### Build log

```
[...]
mkdir /nix/store/yl0rnsdx4ydpaknz8y5sw0f35a85dqrh-mit-scheme-11.2/share/doc/mit-scheme/mit-scheme-ffi
/nix/store/arq10xrdh6yw3hgxils68cnjm0yr756a-coreutils-9.1/bin/install -c --preserve-timestamps -m 644 mit-scheme-ffi/* /nix/store/yl0rnsdx4ypaknz8y5sw0f35a85dqrh-mit-scheme-11.2/share/doc/mit-scheme/mit-scheme-ffi/.
/nix/store/arq10xrdh6yw3hgxils68cnjm0yr756a-coreutils-9.1/bin/install: cannot stat 'mit-scheme-ffi/*': No such file or directory
make[1]: *** [../make-common:159: install-html] Error 1
make[1]: Leaving directory '/build/mit-scheme-11.2/doc/ffi'
make: *** [Makefile:129: install-html] Error 1
make: Leaving directory '/build/mit-scheme-11.2/doc'
error: builder for '/nix/store/f0jidkm5wr8r9b03k7bpbx2840x1h0i9-mit-scheme-11.2.drv' failed with exit code 2;
       last 10 log lines:
       > texi2any --html --no-split ffi.texinfo
       > rm -rf /nix/store/yl0rnsdx4ydpaknz8y5sw0f35a85dqrh-mit-scheme-11.2/share/doc/mit-scheme/mit-scheme-ffi
       > /nix/store/j80wcbm6bb1nsqsgmniq43rg980p27q9-bash-5.2-p15/bin/bash ../mkinstalldirs /nix/store/yl0rnsdx4ydpaknz8y5sw0f35a85dqrh-mit-cheme-11.2/share/doc/mit-scheme/mit-scheme-ffi
       > mkdir /nix/store/yl0rnsdx4ydpaknz8y5sw0f35a85dqrh-mit-scheme-11.2/share/doc/mit-scheme/mit-scheme-ffi
       > /nix/store/arq10xrdh6yw3hgxils68cnjm0yr756a-coreutils-9.1/bin/install -c --preserve-timestamps -m 644 mit-scheme-ffi/* /nix/store/y0rnsdx4ydpaknz8y5sw0f35a85dqrh-mit-scheme-11.2/share/doc/mit-scheme/mit-scheme-ffi/.
       > /nix/store/arq10xrdh6yw3hgxils68cnjm0yr756a-coreutils-9.1/bin/install: cannot stat 'mit-scheme-ffi/*': No such file or directory
       > make[1]: *** [../make-common:159: install-html] Error 1
       > make[1]: Leaving directory '/build/mit-scheme-11.2/doc/ffi'
       > make: *** [Makefile:129: install-html] Error 1
       > make: Leaving directory '/build/mit-scheme-11.2/doc'
       For full logs, run 'nix log /nix/store/f0jidkm5wr8r9b03k7bpbx2840x1h0i9-mit-scheme-11.2.drv'.
```

### Additional context

I have not seen this build successfully on my machine since before March 18. I imagine there was some wide-reaching change merged in from `staging-next` that caused this.

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.26, NixOS, 23.05 (Stoat), 23.05.20230501.8b3bc69`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by leungbk**
###### Description of changes

Fixes #229784

Related to #230712 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -15739,7 +15739,10 @@ with pkgs;
   mitama-cpp-result = callPackage ../development/libraries/mitama-cpp-result { };
 
   mitscheme = callPackage ../development/compilers/mit-scheme
-    { texLive = texlive.combine { inherit (texlive) scheme-small epsf texinfo; }; };
+    {
+      texLive = texlive.combine { inherit (texlive) scheme-small epsf texinfo; };
+      texinfo = texinfo6;
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:15744]: Should we use a different version of texinfo in texlive?

Comment (leungbk) [reply pkgs/top-level/all-packages.nix:15744]: It looks like texlive has only one version of texinfo: https://github.com/nixos/nixpkgs/blob/edefe53eaaa277c8b5f9598b912f13690f7274de/pkgs/tools/typesetting/tex/texlive/tlpdb.nix#L36105-L36111

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -15739,7 +15739,10 @@ with pkgs;
   mitama-cpp-result = callPackage ../development/libraries/mitama-cpp-result { };
 
   mitscheme = callPackage ../development/compilers/mit-scheme
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:15741]:
```suggestion
  mitscheme = callPackage ../development/compilers/mit-scheme {
```

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 230642` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>mitscheme</li>
    <li>mitschemeX11</li>
  </ul>
</details>

**Comment by Mic92** [comment]
Interesting. This built for me locally

**Comment by RaitoBezarius** [comment]
Invalid outputs something, I just throw our remote builders to it

**Comment by Mic92** [comment]
Yeah it used the remote builder that has your name.

**Comment by Mic92** [comment]
Given that it transitioned from a broken state to a broken state, it's not a big loss.

**Comment by Mic92** [comment]
We can wait for hydra says.

**Comment by RaitoBezarius** [comment]
> Maybe a build concurrency issue? 128 build threads here.

Excellent if this is the case.

**Pull request merged**